### PR TITLE
Tidy up checkstyle and similar code quality issues in WebDavStore

### PR DESCRIPTION
**Pull request by philipwhiuk**

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java [hunk 1]
```diff
@@ -47,6 +61,7 @@
  * and email information.
  * </pre>
  */
+@SuppressWarnings("deprecation")
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: I'm still not a fan of this. If we're using deprecated stuff we want to see the warning. Getting rid of the deprecated stuff is the way to get to 0 warnings, not disabling warnings :)

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java [hunk 2]
```diff
@@ -307,50 +332,44 @@ public boolean isCopyCapable() {
...
-        StringBuilder builder = new StringBuilder(200);
-        builder.append("<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>");
-        builder.append("<propfind xmlns=\"DAV:\">");
-        builder.append("<prop>");
-        builder.append("<").append(WebDavConstants.DAV_MAIL_INBOX_FOLDER).append(" xmlns=\"urn:schemas:httpmail:\"/>");
-        builder.append("<").append(WebDavConstants.DAV_MAIL_DRAFTS_FOLDER).append(" xmlns=\"urn:schemas:httpmail:\"/>");
-        builder.append("<").append(WebDavConstants.DAV_MAIL_OUTBOX_FOLDER).append(" xmlns=\"urn:schemas:httpmail:\"/>");
-        builder.append("<").append(WebDavConstants.DAV_MAIL_SENT_FOLDER).append(" xmlns=\"urn:schemas:httpmail:\"/>");
-        builder.append("<").append(WebDavConstants.DAV_MAIL_TRASH_FOLDER).append(" xmlns=\"urn:schemas:httpmail:\"/>");
+        String builder = "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>" +
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: The variable name should now probably be something like `specialFoldersListRequest`.

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java (hunk 1 above)
Comment (philipwhiuk) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: Do we already have an issue to switch to OkHttp - that's really the only way forward here..

Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: Not that I know of. Please feel free to create one.

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java [hunk 3]
```diff
@@ -28,17 +35,24 @@
 import javax.xml.parsers.SAXParser;
 import javax.xml.parsers.SAXParserFactory;
 
-import java.io.*;
+import java.io.BufferedReader;
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: Our [code style settings](https://github.com/k9mail/k-9/wiki/CodeStyle) should place the `java…` imports at the top.

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java [hunk 4]
```diff
@@ -307,50 +332,44 @@ public boolean isCopyCapable() {
...
+            "<prop>" +
+            "<" + WebDavConstants.DAV_MAIL_INBOX_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "<" + WebDavConstants.DAV_MAIL_DRAFTS_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "<" + WebDavConstants.DAV_MAIL_OUTBOX_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "<" + WebDavConstants.DAV_MAIL_SENT_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "<" + WebDavConstants.DAV_MAIL_TRASH_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "<" + WebDavConstants.DAV_MAIL_SPAM_FOLDER + " xmlns=\"urn:schemas:httpmail:\"/>" +
+            "</prop>" +
+            "</propfind>";
         // This should always be ##DavMailSubmissionURI## for which we already have a constant
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: This comment is now in the wrong location.

**Pull request merged**